executor/memtable_reader.go

```diff
@@ -997,3 +997,116 @@ func (e *hotRegionsHistoryRetriver) getHotRegionRowWithSchemaInfo(
 	}
 	return row, nil
 }
+
+type tikvRegionPeersRetriever struct {
+	dummyCloser
+	extractor *plannercore.TikvRegionPeersExtractor
+	retrieved bool
+}
+
+func (e *tikvRegionPeersRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	if e.extractor.SkipRequest || e.retrieved {
+		return nil, nil
+	}
+	e.retrieved = true
```

Comment:
We should change it to true when it is actually retrieved

Reply:
The position of `clusterServerInfoRetriever`  is at the begining too. More specifically, `tikvRegionsPeersRetriver`'s implementation has many return cases, it would be redundant if put `e.retrieved = true` before returns.